apps/sugarfoot/src/workspaces/workspaces.controller.ts

```diff
@@ -21,12 +21,17 @@ import { UpdateWorkspaceDto } from './dto/update-workspace.dto';
 import { AddMemberDto } from './dto/add-member.dto';
 import { JwtAuthGuard } from '@fubs/shared';
 import type { AuthenticatedRequest } from '@fubs/shared';
-import { WorkspaceOwnerGuard } from '../auth/guards/workspace-owner.guard';
 import { WorkspaceMemberGuard } from '../auth/guards/workspace-member.guard';
+import { WorkspacePermissionsByRoleControlGuard } from '../auth/guards/workspace-permissions-by-role.guard';
+import { WorkspacePrivilegesGuard } from '../auth/guards/workspace-privileges.guard';
 
-@UseGuards(JwtAuthGuard)
 @ApiTags('workspaces')
 @ApiBearerAuth()
+@UseGuards(
+  JwtAuthGuard,
+  WorkspacePermissionsByRoleControlGuard,
+  WorkspacePrivilegesGuard
```

Comment:
- JwtAuthGuard = authentication 
- WorkspacePermissionsByRoleControlGuard = check if the user is allowed to act on that HTTP verb by role
- WorkspacePrivilegesGuard = check if the user has access to that workspace